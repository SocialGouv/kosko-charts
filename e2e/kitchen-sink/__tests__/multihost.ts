//
import { config } from "dotenv";
import { KOSKO_BIN, TIMEOUT } from "e2e/templates/helpers";
// eslint-disable-next-line import/default
import execa from "execa";
import { resolve } from "path";

//

const cwd = resolve(__dirname, "..");

test(
  "kosko generate --env dev : should generate a kitchen sink dev manifest",
  async () => {
    const gitlabEnv = config({
      path: resolve(cwd, "./environments/.gitlab.env"),
    }).parsed;
    // Required to allow seemless integration code example
    const result = await execa.node(KOSKO_BIN, ["generate", "--env", "dev"], {
      cwd,
      env: { ...gitlabEnv },
    });

    expect(result.stdout).toMatchSnapshot();
    expect(result.exitCode).toEqual(0);
  },
  TIMEOUT
);
