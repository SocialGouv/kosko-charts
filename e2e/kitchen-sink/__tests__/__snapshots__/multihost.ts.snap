// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`kosko generate --env dev : should generate a kitchen sink dev manifest 1`] = `
"---
apiVersion: v1
kind: Namespace
metadata:
  annotations:
    field.cattle.io/creatorId: gitlab
    field.cattle.io/projectId: c-bd7z2:p-7ms8p
    git/branch: e2e-branch
    git/remote: git@github.com:SocialGouv/sample-kosko.git
    app.gitlab.com/app: socialgouv-sample-kosko
    app.gitlab.com/env: e2e-branch-42
    app.gitlab.com/env.name: e2e-branch-dev2
  labels:
    azure-pg-admin-user: sample-kosko
    application: e2e-branch-42-sample-kosko
    owner: sample-kosko
    team: sample-kosko
    cert: wildcard
  name: sample-kosko-24-e2e-branch-42
---
apiVersion: apps/v1
kind: Deployment
metadata:
  annotations:
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.gitlab.com/app: socialgouv-sample-kosko
    app.gitlab.com/env: e2e-branch-42
    app.gitlab.com/env.name: e2e-branch-dev2
  labels:
    app: hasura
    application: e2e-branch-42-sample-kosko
    owner: sample-kosko
    team: sample-kosko
    cert: wildcard
  name: hasura
  namespace: sample-kosko-24-e2e-branch-42
spec:
  replicas: 1
  selector:
    matchLabels:
      app: hasura
  template:
    metadata:
      annotations:
        kapp.k14s.io/disable-default-ownership-label-rules: ''
        kapp.k14s.io/disable-default-label-scoping-rules: ''
        app.gitlab.com/app: socialgouv-sample-kosko
        app.gitlab.com/env: e2e-branch-42
        app.gitlab.com/env.name: e2e-branch-dev2
      labels:
        app: hasura
        application: e2e-branch-42-sample-kosko
        owner: sample-kosko
        team: sample-kosko
        cert: wildcard
    spec:
      containers:
        - image: hasura/graphql-engine:latest
          livenessProbe:
            failureThreshold: 6
            httpGet:
              path: /healthz
              port: http
            initialDelaySeconds: 30
            periodSeconds: 5
            timeoutSeconds: 5
          name: hasura
          ports:
            - containerPort: 80
              name: http
          readinessProbe:
            failureThreshold: 15
            httpGet:
              path: /healthz
              port: http
            initialDelaySeconds: 0
            periodSeconds: 5
            successThreshold: 1
            timeoutSeconds: 1
          resources:
            limits:
              cpu: 500m
              memory: 512Mi
            requests:
              cpu: 100m
              memory: 64Mi
          startupProbe:
            failureThreshold: 12
            httpGet:
              path: /healthz
              port: http
            periodSeconds: 5
          envFrom:
            - secretRef:
                name: azure-pg-user-8843083
      imagePullSecrets:
        - name: regcred
      initContainers:
        - env:
            - name: WAIT_FOR_RETRIES
              value: '24'
          envFrom:
            - secretRef:
                name: azure-pg-user-8843083
          image: ghcr.io/socialgouv/docker/wait-for-postgres:6.32.2
          imagePullPolicy: Always
          name: wait-for-postgres
          resources:
            limits:
              cpu: 20m
              memory: 32Mi
            requests:
              cpu: 5m
              memory: 16Mi
---
apiVersion: v1
kind: Service
metadata:
  labels:
    app: hasura
    application: e2e-branch-42-sample-kosko
    owner: sample-kosko
    team: sample-kosko
    cert: wildcard
  name: hasura
  annotations:
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.gitlab.com/app: socialgouv-sample-kosko
    app.gitlab.com/env: e2e-branch-42
    app.gitlab.com/env.name: e2e-branch-dev2
  namespace: sample-kosko-24-e2e-branch-42
spec:
  ports:
    - name: http
      port: 80
      targetPort: 80
  selector:
    app: hasura
  type: ClusterIP
---
apiVersion: apps/v1
kind: Deployment
metadata:
  annotations:
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.gitlab.com/app: socialgouv-sample-kosko
    app.gitlab.com/env: e2e-branch-42
    app.gitlab.com/env.name: e2e-branch-dev2
  labels:
    app: www
    application: e2e-branch-42-sample-kosko
    owner: sample-kosko
    team: sample-kosko
    cert: wildcard
    component: next
  name: www
  namespace: sample-kosko-24-e2e-branch-42
spec:
  replicas: 1
  selector:
    matchLabels:
      app: www
  template:
    metadata:
      annotations:
        kapp.k14s.io/disable-default-ownership-label-rules: ''
        kapp.k14s.io/disable-default-label-scoping-rules: ''
        app.gitlab.com/app: socialgouv-sample-kosko
        app.gitlab.com/env: e2e-branch-42
        app.gitlab.com/env.name: e2e-branch-dev2
      labels:
        app: www
        application: e2e-branch-42-sample-kosko
        owner: sample-kosko
        team: sample-kosko
        cert: wildcard
        component: next
    spec:
      containers:
        - image: >-
            registry.gitlab.factory.social.gouv.fr/socialgouv/sample-kosko/www:8843083edb7f873cad1d1420731a60773594ffae
          livenessProbe:
            failureThreshold: 6
            httpGet:
              path: /healthz
              port: http
            initialDelaySeconds: 30
            periodSeconds: 5
            timeoutSeconds: 5
          name: www
          ports:
            - containerPort: 8080
              name: http
          readinessProbe:
            failureThreshold: 15
            httpGet:
              path: /healthz
              port: http
            initialDelaySeconds: 0
            periodSeconds: 5
            successThreshold: 1
            timeoutSeconds: 1
          resources:
            limits:
              cpu: 50m
              memory: 128Mi
            requests:
              cpu: 1m
              memory: 64Mi
          startupProbe:
            failureThreshold: 12
            httpGet:
              path: /healthz
              port: http
            periodSeconds: 5
          envFrom:
            - secretRef:
                name: azure-pg-user-8843083
      imagePullSecrets:
        - name: some-secret
      initContainers:
        - env:
            - name: WAIT_FOR_RETRIES
              value: '24'
          envFrom:
            - secretRef:
                name: azure-pg-user-8843083
          image: ghcr.io/socialgouv/docker/wait-for-postgres:6.32.2
          imagePullPolicy: Always
          name: wait-for-postgres
          resources:
            limits:
              cpu: 20m
              memory: 32Mi
            requests:
              cpu: 5m
              memory: 16Mi
---
apiVersion: v1
kind: Service
metadata:
  labels:
    app: www
    application: e2e-branch-42-sample-kosko
    owner: sample-kosko
    team: sample-kosko
    cert: wildcard
  name: www
  annotations:
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.gitlab.com/app: socialgouv-sample-kosko
    app.gitlab.com/env: e2e-branch-42
    app.gitlab.com/env.name: e2e-branch-dev2
  namespace: sample-kosko-24-e2e-branch-42
spec:
  ports:
    - name: http
      port: 80
      targetPort: 8080
  selector:
    app: www
  type: ClusterIP
---
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  annotations:
    kubernetes.io/ingress.class: nginx
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.gitlab.com/app: socialgouv-sample-kosko
    app.gitlab.com/env: e2e-branch-42
    app.gitlab.com/env.name: e2e-branch-dev2
  labels:
    app: www
    application: e2e-branch-42-sample-kosko
    owner: sample-kosko
    team: sample-kosko
    cert: wildcard
  name: www
  namespace: sample-kosko-24-e2e-branch-42
spec:
  rules:
    - host: e2e-branch-42-sample-kosko.dev2.fabrique.social.gouv.fr
      http:
        paths:
          - backend:
              service:
                name: www
                port:
                  name: http
            path: /
            pathType: Prefix
  tls:
    - hosts:
        - e2e-branch-42-sample-kosko.dev2.fabrique.social.gouv.fr
      secretName: wildcard-crt
---
apiVersion: autoscaling/v2beta2
kind: HorizontalPodAutoscaler
metadata:
  name: www
  namespace: sample-kosko-24-e2e-branch-42
spec:
  maxReplicas: 10
  metrics:
    - resource:
        name: cpu
        target:
          averageUtilization: 100
          type: Utilization
      type: Resource
    - resource:
        name: memory
        target:
          averageUtilization: 100
          type: Utilization
      type: Resource
  minReplicas: 1
  scaleTargetRef:
    apiVersion: apps/v1
    kind: Deployment
    name: www"
`;
